src/main/scala/com/danielasfregola/twitter4s/http/clients/streaming/StreamingClientNew.scala

```diff
@@ -0,0 +1,76 @@
+package com.danielasfregola.twitter4s.http.clients.streaming
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import akka.stream.scaladsl.{Framing, Source}
+import akka.stream.{ActorMaterializer, SharedKillSwitch}
+import akka.util.ByteString
+import com.danielasfregola.twitter4s.entities.streaming.StreamingMessage
+import com.danielasfregola.twitter4s.entities.{AccessToken, ConsumerToken}
+import com.danielasfregola.twitter4s.exceptions.TwitterException
+import com.danielasfregola.twitter4s.http.clients.OAuthClient
+import com.danielasfregola.twitter4s.http.oauth.OAuth1Provider
+import org.json4s.native.Serialization
+
+import scala.concurrent.Future
+import scala.util.Try
+
+private[twitter4s] class StreamingClientNew(val consumerToken: ConsumerToken, val accessToken: AccessToken)(
+    implicit val system: ActorSystem)
+    extends OAuthClient {
+
+  override def withLogRequest: Boolean = true
+  override def withLogRequestResponse: Boolean = true
+
+  implicit val materializer = ActorMaterializer()
+  implicit val ec = materializer.executionContext
+
+  def preProcessing(): Unit = ()
+
+  lazy val oauthProvider = new OAuth1Provider(consumerToken, Some(accessToken))
+
+  def getStream(request: HttpRequest, killSwitch: SharedKillSwitch): Future[Source[StreamingMessage, NotUsed]] = {
```

Comment:
I finally have some spare time to look at this!

I am struggling to understand `Future[Source[StreamingMessage, NotUsed]]`, in particular the `NotUsed` part. Could you please clarify your reasoning?

Reply:
Certainly, the second type parameter is required by akka streams. It can be used to extract additional materialized value from each step. `NotUsed` is the default type, and is used here because we don't need any additional value other than `StreamingMessage`. But this version is already slightly outdated, let me update my PR